### PR TITLE
UI components comparison

### PR DESCRIPTION
**Pull request by espinr**
A new table with an analysis that compares how the different MiniApp versions implement their MiniApp Components. It will complement the [existing comparison table](https://w3c.github.io/miniapp-white-paper/comparison.html), but focused on components.

Preview: https://raw.githack.com/espinr/miniapp-white-paper/ui-components-comparison/components-comparison.html

### CODE REVIEW

**Comment by xfq** [comment]
Since [PR Preview](https://github.com/tobie/pr-preview) does not support multiple HTML files and does not support copying images and spec-specific stylesheets/scripts into the preview, I added Netlify to this repo for the preview function.

**Comment by xfq** [comment]
We should link to this table somewhere in the white paper.

File: components-comparison.html [hunk 1]
```diff
@@ -0,0 +1,1484 @@
...
+                </tr>
+                <tr>
+                    <td class="c0">Built-in <abbr title="Components">Comp.</abbr></td>
+                    <td class="c1">checkbox</td>
+                    <td class="c2">
+                        <details>
+                            <summary>
+                                <code>&lt;input type="checkbox"&gt;</code>
+                            </summary>
+                            <code>&lt;input type="checkbox"&gt;</code>, OpenUI's <code>  &lt;checkbox&gt;</checkbox>
```
Comment (xfq) [inline components-comparison.html]:
```suggestion
                            <code>&lt;input type="checkbox"&gt;</code>, OpenUI's <code>&lt;checkbox&gt;</code>
```

File: components-comparison.html [hunk 2]
```diff
@@ -0,0 +1,1484 @@
...
+                                <pre>
+&lt;view&gt; {{ message }} &lt;/view&gt;
+
+Page({ 
+   data: { 
+     message: 'Hello alipay!', 
+    }, 
+});
+                                </pre>  
+                            </code>
```
Comment (xfq) [inline components-comparison.html]:
```suggestion
                            <pre><code>&lt;view&gt; {{ message }} &lt;/view&gt;

Page({ 
   data: { 
     message: 'Hello alipay!', 
    }, 
});</code></pre>
```

Should be `<pre><code>` per https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-code-element

Comment (xfq) [reply components-comparison.html]: Same below.

File: components-comparison.html [hunk 3]
```diff
@@ -0,0 +1,1484 @@
...
+                    <td class="c2 partial"><a href="https://open-ui.org/components/tabs.research">OpenUI research</a></td>
+                    <td class="c3">
+                        <details>
+                            <summary>
+                                <code>&lt;tabs&gt;</code>
+                            </summary>
+                            <code>&lt;tabs&gt;</code>, <code>&lt;tab-bar&gt;</code>, <code>&lt;tab-content&gt;</code>
+                        </details>
+                    </td>
+                    <td class="c4 no"></td>
```
Comment (xfq) [inline components-comparison.html]: The UI component framework for Alipay created by Ant Group—Ant Design Mini—supports tabs, though: https://mini.ant.design/components/tabs

File: components-comparison.html [hunk 4]
```diff
@@ -0,0 +1,1484 @@
...
+                    <td class="c1">rating</td>
+                    <td class="c2 no"></td>
+                    <td class="c3"><code>&lt;rating&gt;</code></td>
+                    <td class="c4 no"></td>
+                    <td class="c5 no"></td>
+                </tr>
+                <tr>
+                    <td class="c0">Built-in <abbr title="Components">Comp.</abbr></td>
+                    <td class="c1">marquee</td>
+                    <td class="c2 no"></td>
```
Comment (xfq) [inline components-comparison.html]: There's https://html.spec.whatwg.org/multipage/rendering.html#the-marquee-element-2

File: index.html [hunk 5]
```diff
@@ -251,6 +251,9 @@ <h2>Gap between MiniApps and PWA</h2>
                     </tr>
                   </tbody>
                   </table>
+                <p>
+                  Another table in the annex summarizes the <a href="components-comparison.html">differences between MiniApp components and Web Components</a>.
```
Comment (xfq) [inline index.html]: "Web Components" may not be the right term here, as it usually refers to custom elements. Maybe just "HTML"?

File: index.html [hunk 6]
```diff
@@ -995,7 +998,7 @@ <h2>
         Gap Analysis
       </h2>
       <p>
-        Please look at <a href="comparison.html">the comparison table of APIs in MiniApps, W3C specs, and PWAs</a>.
+        Learn more about the <a href="comparison.html">differences between MiniApp and PWA</a>, considering the status of the W3C specifications. Also, find a summary of the <a href="components-comparison.html">differences between MiniApp components</a> in the different MiniApp versions compared to Web Components.
```
Comment (xfq) [inline index.html]: "MiniApp user agents / implementations" might be more appropriate than "MiniApp versions" here.

File: index.html (hunk 5 above)
Comment (espinr) [reply index.html]: Right. I've changed the term to **standard HTML**.

Comment (espinr) [reply index.html]: Changed here: c41398961aa698e402d5396e318ee5743791060d

File: index.html (hunk 6 above)
Comment (espinr) [reply index.html]: Good point, thanks. Changed at 4994bb011c7f832a6346f6fdaf6eb122c63fbffe

File: components-comparison.html (hunk 4 above)
Comment (espinr) [reply components-comparison.html]: Thanks @xfq . My brain omitted it. I remember  using it :) Changed at 48b71421bad355590535d06c68b4187e938be234

File: components-comparison.html (hunk 3 above)
Comment (espinr) [reply components-comparison.html]: Thanks for the pointer. Fixed: 055e12c06d909bd52f78a7957ffbf0737d558a9c

**Comment by espinr** [comment]
Thanks for your comprehensive review @xfq

**Pull request merged**